compiler/ilgen/OMRVirtualMachineRegisterInStruct.hpp

```diff
@@ -71,16 +71,16 @@ class VirtualMachineRegisterInStruct : public TR::VirtualMachineRegister
       _fieldName(fieldName),
       _localNameHoldingStructAddress(localNameHoldingStructAddress)
       {
-      _integerTypeForAdjustments = b->typeDictionary()->GetFieldType(structName, fieldName);
-      if (_integerTypeForAdjustments->isPointer())
+      TR::IlType *registerBaseType = b->typeDictionary()->GetFieldType(structName, fieldName);
+      if (registerBaseType->isPointer())
          {
-         _integerTypeForAdjustments = _integerTypeForAdjustments->baseType();
-         if (_integerTypeForAdjustments->isPointer())
+         registerBaseType = registerBaseType->baseType();
+         if (registerBaseType->isPointer())
```

Comment:
I don't think this should dig into two levels of pointer here: it starts from the field type, which should only require one level of pointer to primitive type to do the right thing (i.e. if it's a pointer type use Word otherwise use the field's primitive type).

Reply:
It's possible this change may induce some required changes in our samples, but I would hope not.